packages/app-accounts/src/UploadModal.tsx

```diff
@@ -0,0 +1,120 @@
+// Copyright 2017-2018 @polkadot/app-accounts authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { I18nProps, BareProps } from '@polkadot/ui-app/types';
+import { KeyringPair$Json } from '@polkadot/util-keyring/types';
+
+import React from 'react';
+
+import Button from '@polkadot/ui-app/Button';
+import Modal from '@polkadot/ui-app/Modal';
+import classes from '@polkadot/ui-app/util/classes';
+import keyring from '@polkadot/ui-keyring/index';
+import Unlock from '@polkadot/ui-signer/Unlock';
+import isUndefined from '@polkadot/util/is/undefined';
+
+import translate from './translate';
+
+type Props = I18nProps & BareProps & {
+  address?: string,
+  error?: React.ReactNode,
+  hidePasswordModal: () => void,
+  isPasswordModalOpen: boolean,
+  onChangePassword: (password: string) => void,
+  onDiscard: () => void,
+  password: string,
+  processUploadedFileStorage: () => void,
+  uploadedFileKeyringPair?: KeyringPair$Json
+};
+
+class UploadModal extends React.PureComponent<Props> {
+  render () {
+    const { address, className, hidePasswordModal, isPasswordModalOpen, style, t } = this.props;
+
+    return (
+      <Modal
+        className={classes('ui--accounts-upload-Signer', className)}
```

Comment:
I'm assuming that we're using BEM http://getbem.com/naming/. 
I used the `-Signer` postfix in UploadModal so it would be consistent with the naming convention used in ui-signer's Modal. I'll remove the `ui--`prefix as you've suggested. I'm going to propose to make it `accounts--UploadModal-Signer` (since that'd be consistent with how it's done everywhere else). I'll make wrapper div in UploadButton `accounts--UploadButton`